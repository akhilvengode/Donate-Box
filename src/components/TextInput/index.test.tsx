import { shallow } from "enzyme";
import Input from ".";

describe("Testing input component", () => {
  const onChangeFunction = jest.fn();
  const wrapper = shallow(<Input id="test" onChange={onChangeFunction} />);

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the input component", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("should call the given function on onChange event happens", () => {
    wrapper.find("#test").simulate("change");
    expect(onChangeFunction).toBeCalled();
  });

  it("should render error message when error is set", () => {
    const wrapper2 = shallow(
      <Input id="test" onChange={onChangeFunction} error errorMessage="Error" />
    );
    expect(wrapper2.find("p").text()).toEqual("Error");
  });
});
