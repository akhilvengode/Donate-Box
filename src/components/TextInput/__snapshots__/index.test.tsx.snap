// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing input component should match the snapshot 1`] = `
<div
  className="input"
>
  <label
    className="input__label"
    htmlFor="test"
  />
  <input
    id="test"
    onChange={[MockFunction]}
    type="text"
  />
</div>
`;
